fairroot/geobase/FairGeoSet.cxx

```diff
@@ -517,3 +518,30 @@ void FairGeoSet::compare(FairGeoSet& rset)
     }
     cout << "Number of additional volumes in second list: " << nnf << '\n';
 }
+
+const char* FairGeoSet::getModuleName(Int_t m)
+{
+    /** Returns the module name of the detector with array index m
+        Since the name uses natuaral counting the index has to be
+        increased by 1. All module names in the ASCII file needs to
+        start with the name saved in fName since otherwise they are
+        not constructed.
+    */
+    assert(m >= 0);
```

Comment:
Needs `#include <cassert>` (https://en.cppreference.com/w/cpp/error/assert)

Reply:
Thanks for catching this. If used now the fairlogger LOG_IF instead of assert.

Reply:
But `LOG_IF` you always pay for even in Release builds? The behaviour of `assert` is that it becomes a no-op with `-DNDEBUG`.

Reply:
The problem I see is that the files passed to the ASCII interface are a runtime dependency and can be changed anytime. So the check if the format is correct should be done in any case and the execution should stop if a problem is detected. In my opinion `LOG_IF` is the proper solution here.